programs/margin-account/src/lib.rs

```diff
@@ -64,26 +71,42 @@ pub struct Withdraw<'info> {
 }
 
 #[derive(Accounts)]
-pub struct Trade<'info> {
+pub struct OpenPosition<'info> {
     // TODO
     #[account(signer)]
     authority: AccountInfo<'info>,
 }
 
+#[derive(Accounts)]
+pub struct ClosePosition<'info> {
+    // TODO
+    #[account(signer)]
+    authority: AccountInfo<'info>,
+}
+
+//? Possibly add cancel position, if it cannot be combined with close.
+
 #[derive(Accounts)]
 pub struct Liquidate<'info> {
     // TODO
     authority: AccountInfo<'info>,
 }
 
-/// Margin account which handles
+/// Margin account state which keeps track of positions opened for a given trader.
 #[account]
 pub struct MarginAccount {
     /// The owner of this margin account.
     pub trader: Pubkey,
-    /// Address of the account's token vault.
-    pub vault: Pubkey,
-    /// Signer nonce.
-    pub nonce: u8,
```

Comment:
Most likely we will need to bring this back. Will do it in my PR on the flow of things.

Reply:
I thought the vault had to be passed in, because otherwise it would be statically pointing at one denom (when the margin account needs to handle multiple). Sounds good though, feel free to add back when we need